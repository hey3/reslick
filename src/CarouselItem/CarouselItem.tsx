import React from 'react'
import styled from 'styled-components'

type ContainerProps = React.ImgHTMLAttributes<HTMLImageElement> & {
  className?: string
  /** href for component */
  href?: string
}
type PresenterProps = Record<string, unknown>
type Props = ContainerProps & PresenterProps

const DomComponent: React.FC<Props> = ({ className, href, alt, children, ...props }) => {
  return (
    <li className={`${className} carousel-item-root`}>
      <a className="carousel-link-area" href={href} aria-label={props['aria-label']}>
        <figure className="carousel-image-are">
          <picture className="carousel-picture">
            <img className="carousel-image" loading="lazy" alt={alt} {...props} />
          </picture>
        </figure>
      </a>
      <div className="carousel-content">{children}</div>
    </li>
  )
}

const StyledComponent = styled(DomComponent)`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  gap: calc(2.5rem / 2);
  scroll-snap-align: start;

  a {
    align-self: center;

    & > figure {
      position: relative;
      justify-content: center;
      width: inherit;
      height: inherit;
      inline-size: inherit;
      block-size: inherit;
      margin: 0;
      cursor: pointer;
      user-select: none;

      & > picture {
        display: inline-block;
        width: inherit;
        height: inherit;
        inline-size: inherit;
        block-size: inherit;
        border-radius: 1ex;
        box-shadow: 0 1px 2px 0 hsl(0 0% 0% / 20%);
        overflow: hidden;

        & > img {
          display: block;
          width: 100%;
          height: 100%;
          inline-size: 100%;
          block-size: 100%;
          background-image: linear-gradient(to bottom, hsl(210 10% 40%), hsl(210 10% 20%));
          background-color: #525252;
          object-fit: cover;
        }
      }
    }
  }
`

export const CarouselItem: React.FC<ContainerProps> = ({ children, ...props }) => {
  return <StyledComponent {...props}>{children}</StyledComponent>
}
