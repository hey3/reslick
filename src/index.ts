import { Carousel } from './Carousel'
import { CarouselItem } from './CarouselItem'

export default Carousel

export { Carousel, CarouselItem }
