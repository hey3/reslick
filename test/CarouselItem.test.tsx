import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import { CarouselItem } from '../src/CarouselItem'

describe('CarouselItem', () => {
  it('renders correctly', () => {
    const defaultWrapper = renderer
      .create(
        <CarouselItem />
      )
      .toJSON()
    expect(defaultWrapper).toMatchSnapshot()

    const withContentWrapper = renderer.create(
      <CarouselItem>
        <article>
          <h1>test title</h1>
          <p>test</p>
        </article>
      </CarouselItem>
    )
    expect(withContentWrapper).toMatchSnapshot()
  })
})
