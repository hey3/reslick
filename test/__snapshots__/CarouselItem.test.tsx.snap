// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarouselItem renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  gap: calc(2.5rem / 2);
  -webkit-scroll-snap-align: start;
  -moz-scroll-snap-align: start;
  -ms-scroll-snap-align: start;
  scroll-snap-align: start;
}

.c0 a {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c0 a > figure {
  position: relative;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: inherit;
  height: inherit;
  inline-size: inherit;
  -webkit-block-size: inherit;
  -ms-flex-block-size: inherit;
  block-size: inherit;
  margin: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 a > figure > picture {
  display: inline-block;
  width: inherit;
  height: inherit;
  inline-size: inherit;
  -webkit-block-size: inherit;
  -ms-flex-block-size: inherit;
  block-size: inherit;
  border-radius: 1ex;
  box-shadow: 0 1px 2px 0 hsl(0 0% 0% / 20%);
  overflow: hidden;
}

.c0 a > figure > picture > img {
  display: block;
  width: 100%;
  height: 100%;
  inline-size: 100%;
  -webkit-block-size: 100%;
  -ms-flex-block-size: 100%;
  block-size: 100%;
  background-image: linear-gradient(to bottom,hsl(210 10% 40%),hsl(210 10% 20%));
  background-color: #525252;
  object-fit: cover;
}

<li
  className="c0 carousel-item-root"
>
  <a
    className="carousel-link-area"
  >
    <figure
      className="carousel-image-are"
    >
      <picture
        className="carousel-picture"
      >
        <img
          className="carousel-image"
          loading="lazy"
        />
      </picture>
    </figure>
  </a>
  <div
    className="carousel-content"
  />
</li>
`;

exports[`CarouselItem renders correctly 2`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  gap: calc(2.5rem / 2);
  -webkit-scroll-snap-align: start;
  -moz-scroll-snap-align: start;
  -ms-scroll-snap-align: start;
  scroll-snap-align: start;
}

.c0 a {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c0 a > figure {
  position: relative;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: inherit;
  height: inherit;
  inline-size: inherit;
  -webkit-block-size: inherit;
  -ms-flex-block-size: inherit;
  block-size: inherit;
  margin: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 a > figure > picture {
  display: inline-block;
  width: inherit;
  height: inherit;
  inline-size: inherit;
  -webkit-block-size: inherit;
  -ms-flex-block-size: inherit;
  block-size: inherit;
  border-radius: 1ex;
  box-shadow: 0 1px 2px 0 hsl(0 0% 0% / 20%);
  overflow: hidden;
}

.c0 a > figure > picture > img {
  display: block;
  width: 100%;
  height: 100%;
  inline-size: 100%;
  -webkit-block-size: 100%;
  -ms-flex-block-size: 100%;
  block-size: 100%;
  background-image: linear-gradient(to bottom,hsl(210 10% 40%),hsl(210 10% 20%));
  background-color: #525252;
  object-fit: cover;
}

<li
  className="c0 carousel-item-root"
>
  <a
    className="carousel-link-area"
  >
    <figure
      className="carousel-image-are"
    >
      <picture
        className="carousel-picture"
      >
        <img
          className="carousel-image"
          loading="lazy"
        />
      </picture>
    </figure>
  </a>
  <div
    className="carousel-content"
  >
    <article>
      <h1>
        test title
      </h1>
      <p>
        test
      </p>
    </article>
  </div>
</li>
`;
